lib/cppcheck.cpp

```diff
@@ -131,16 +132,36 @@ namespace {
                     args += " " + v.get<std::string>();
             }
 
+            if(obj.count("python")) {
+                // Python was defined in the config file
+                if (obj["python"].is<picojson::array>()){
+                    return "Loading" + fileName +"failed. python must not be an array.";
+                }
+                python = obj["python"].get<std::string>();
+            }
+            else{
+                python = "";
+            }
+
             return getAddonInfo(obj["script"].get<std::string>(), exename);
         }
     };
 }
 
-static std::string executeAddon(const AddonInfo &addonInfo, const std::string &dumpFile)
-{
+static std::string executeAddon(const AddonInfo &addonInfo,
+                                const std::string &defaultPythonExe,
+                                const std::string &dumpFile) {
+
+    std::string pythonExe = (addonInfo.python != "") ? addonInfo.python : defaultPythonExe;
+
+    if(pythonExe.find(" ") != std::string::npos){
+        // popen strips the first quote. Needs 2 sets to fully quote.
```

Comment:
Interesting comment, I did not know that. Strange..

Reply:
Yes it took me by surprise as well. The initial set of " was always being removed, so paths with a space in them were not working. I make a habit of keeping my paths space-free so ive never encountered a popen problem like that before.